.github/workflows/ci-test.yml

```diff
@@ -40,11 +40,48 @@ jobs:
 
       # run with sudo (...) --privileged
       # this is needed to create network namespaces for the ebpf tests
-    - name: Run tests (Linux)
+    - name: Run tests (Ubuntu Linux)
       run: |
         sudo docker run --privileged -w /p4c/build -e $CTEST_PARALLEL_LEVEL p4c ctest --output-on-failure --schedule-random
       if: matrix.unified == 'ON'
 
+  # Build and test p4c on Fedora
+  build-fedora-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        unified: [ON, OFF]
+        enable_gmp: [ON, OFF]
+        exclude:
+          - unified: OFF
+            enable_gmp: OFF
+    runs-on: ubuntu-latest
```

Comment:
I really do not know the format of these files, so this is just a visual "is this what you meant to do?" that could be perfectly correct.  Just surprised me to see build-fedora-linux running on ubuntu-latest

Reply:
Hi Andy, thank you for asking. I have to admit that it looks confusing and it would be good to add a comment here.

GitHub supports a very limited set of [virtual environments](https://github.com/actions/virtual-environments). The following means that the GitHub action would run inside a Fedora container that runs in a Ubuntu VM:

```yml
runs-on: ubuntu-latest
    container:
      image: registry.fedoraproject.org/fedora:latest
```